packages/vue-components/src/Navbar.vue

```diff
@@ -71,12 +75,89 @@
         default:
           return 'navbar-dark bg-primary'
         }
-      }
+      },
     },
     methods: {
       toggleCollapse (e) {
         e && e.preventDefault()
         this.collapsed = !this.collapsed
+      },
+      // add directory index, e.g http://foo/ or http://foo becomes http://foo/index.html
+      normalizeUrl(url) {
+        if (url.endsWith('.html')) {
+          return url;
+        } else if (url.endsWith('/')) {
+          return `${url}index.html`;
+        } else {
+          return `${url}/index.html`;
+        }
+      },
+      highlightLink(url) {
+        // Performs an exact equality match on links
+        if (this.hasLinkMatch(url, true)) {
+          return;
+        }
+        return this.hasLinkMatch(url, false);
+      },
+      hasLinkMatch(url, isStrict) {
+        const navLis = Array.from(this.$el.querySelector('.navbar-nav').children);
+        for (const li of navLis) {
+          // there are two kinds of navlinks, get all
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            const aNorm = this.normalizeUrl(a.href);
+            const urlNorm = this.normalizeUrl(url);
+            if (isStrict) {
+              // checks by strict equality of URL
+              if (aNorm === urlNorm) {
+                li.classList.add('current');
+                return true;
+              }
+            } else {
+              const aTagUrl = new URL(aNorm);
+              const browserUrl = new URL(urlNorm);
+              const aParts = `${aTagUrl.host}${aTagUrl.pathname}`.split('/').slice(1);
+              const bParts = `${browserUrl.host}${browserUrl.pathname}`.split('/').slice(1);
```

Comment:
hmm should we move the browser parts out of the loops? since they're the same for any iteration